### PR TITLE
Make image helper errors fatal 

### LINKED ISSUE #67: Old {{image}} helper is not being detected within a partial file.
I was updating a old theme so that it can be compatible with Ghost 1.0.0. I have found this issue. If it's intended behavior then feel free to close this issue.

### Exact case

I have `{{tag.image}}` within a partial file named **cover-tag.hbs** . Which is being called within **tag.hbs** file.

As we know, In ghost 1.0.0 `{{image}}` or in this case `{{tag/image}}` can not be used. So gscan should show error or warning. But If I upload the theme including this code there is no error or warning. Theme check passed.

**Code in the cover-tag.hbs file**

```
<div class="row cover tag-cover {{#if tag.image}}" style="background-image: url({{tag.image}}) {{else}}no-image{{/if}}">
	<div class="col-md-12 cover-inner">
		<h3 class="tag-name">
			Tag: {{tag.name}}
		</h3>
		<div class="post-count">
			Total {{plural pagination.total empty='0 Post' singular='% Post' plural='% Posts'}}
		</div>
		{{#if tag.description}}
		<div class="tag-description">
	        {{tag.description}}
		</div>
		{{/if}}
	</div>
</div>
```

**Code in tag.hbs file where cover-tag.hbs file is being called**

```

{{!< default}}
{{! The tag above means - insert everything in this file into the {body} of the default.hbs template }}

<div class="content-area-wrap">
	{{> mobile-header}}
	<!-- start main content -->
	<div class="container-fluid main-content-wrap">
		{{> cover-tag}}
		<div class="row posts-wrap-outer">
			{{#foreach posts}}
			<!-- start post -->
			.............................................................................................................
			...........this part of code is not relevant for this issue ...........................
			.............................................................................................................
			<!-- end-post -->
			{{/foreach}}
		</div>
		{{pagination}}
	</div>
	<!-- end main content -->
</div>
```

**Passed when uploading in to Ghost dashboard** 

after upload

**Passed when uploading in to gscan.ghost.com** 

theme check passed

### PR DESCRIPTION
**Pull request by cobbspur**
Attempted to cover the most common use cases of image helpers 

refs #63, closes #67

- Changes errors relating to image helper to fatal errors.
- Allows whitespace and line breaks to be included in checks for deprecated handlebars.
- Fixes regex for tag.image
- Add common nested image regex 
- Add regex checks for image conditionals

### CODE REVIEW

File: lib/checks/001-deprecations.js [hunk 1]
```diff
@@ -5,75 +5,125 @@ var _       = require('lodash'),
...
+            helperName: '{{author.cover}}',
+            ruleCode: 'GS001-DEPR-AC'
+        },
+        {
+            helperRegEx: /{{[\s\S]?tag\.image[\s\S]?}}/g,
+            helperName: '{{tag.image}}',
+            ruleCode: 'GS001-DEPR-TIMG'
+        },
+        {
+            helperRegEx: /{{[\s\S]?post.author\.image[\s\S]?}}/g,
```
Comment (kirrg001) [inline lib/checks/001-deprecations.js]: An escape of the dot is missing. Otherwise it would match any character.

**Comment by cobbspur** [comment]
@kirrg001 passing now. I think the only other case I haven't added yet is {{ tags.[4].image}} from say a post context. I can that and a corresponding conditional in if the rest of this looks sane

File: test/fixtures/themes/001-deprecations/invalid/post.hbs [hunk 2]
```diff
@@ -1,6 +1,30 @@
...
+    {{post.author.cover}}
+{{/if}}
+
+{{#if post.tags.[3].image}}
+    {{post.tags.[3].image}}
+{{/if}}
 {{#post}}
     {{meta_description}}
-    {{image}}
+    {{#if image}}
```
Comment (kirrg001) [inline test/fixtures/themes/001-deprecations/invalid/post.hbs]: How about adding one example with white spaces?

File: lib/checks/001-deprecations.js [hunk 3]
```diff
@@ -5,75 +5,125 @@ var _       = require('lodash'),
...
-            },
-            {
-                cssRegEx: /\.page-template-\w+[\s\{]/g,
-                className: '.page-template-slug',
-                css: true,
-                ruleCode: 'GS001-DEPR-CSS-PATS'
-            }
-        ];
+        {
+            helperRegEx: /{{\s*?pageUrl['"\w\d\s]*?}}/ig,
```
Comment (kirrg001) [inline lib/checks/001-deprecations.js]: This regex also matches {{pageUrlA]]?
But it should only match [[pageUrl something]].

**Pull request merged**